libs/server/AOF/AofHeader.cs

```diff
@@ -5,16 +5,72 @@
 
 namespace Garnet.server
 {
-    [StructLayout(LayoutKind.Explicit, Size = 14)]
+    [StructLayout(LayoutKind.Explicit, Size = 15)]
     struct AofHeader
     {
+        // UPDATE THIS IF YOU CHANGE LAYOUT OF THIS STRUCT, this cannot exceed 127, as long as we are using MSB as heuristic to check for legacy format
+        const byte CURRENT_AOF_FORMAT_VERSION = 1;
+
+        /*
+        We use the MSB to differentiate if we are interacting with a legacy or the new AofHeader  
+        Note: When we are removing legacy support for old AOF log we can remove this and it's usage completely
+        */
+        const byte ONLY_MSB_SET_MASK = 0b1000_0000;
```

Comment:
1 << 7 and 1 - (1 << 7) might be a tad more readable imo. Up to you.

Reply:
keeping it as binary since one byte in binary feels more legible like this to me atm.